src/lib.rs

```diff
@@ -530,45 +539,30 @@ pub(crate) unsafe fn rav1d_parse_sequence_header(
     if !ptr.is_null() {
         res = rav1d_data_wrap_internal(&mut buf, ptr, sz, Some(dummy_free), 0 as *mut c_void);
         if res.is_err() {
-            current_block = 10647346020414903899;
-        } else {
-            current_block = 5399440093318478209;
+            return dav1d_parse_sequence_header_error(res, c, &mut buf);
         }
-    } else {
-        current_block = 5399440093318478209;
     }
-    loop {
-        match current_block {
-            10647346020414903899 => {
-                rav1d_data_unref_internal(&mut buf);
-                break;
-            }
-            _ => {
-                if buf.sz > 0 {
-                    let res = rav1d_parse_obus(c, &mut buf, 1 as c_int);
-                    let Ok(res) = res else {
-                        current_block = 10647346020414903899;
-                        continue;
-                    };
-                    if !(res as usize <= buf.sz) {
-                        unreachable!();
-                    }
-                    buf.sz = (buf.sz as c_ulong).wrapping_sub(res as c_ulong) as usize as usize;
-                    buf.data = (buf.data).offset(res as isize);
-                    current_block = 5399440093318478209;
-                } else if ((*c).seq_hdr).is_null() {
-                    res = Err(ENOENT);
-                    current_block = 10647346020414903899;
-                } else {
-                    *out = (*(*c).seq_hdr).clone();
-                    res = Ok(());
-                    current_block = 10647346020414903899;
-                }
-            }
-        }
+
+    while buf.sz > 0 {
+        let res = rav1d_parse_obus(c, &mut buf, 1 as c_int);
+        let Ok(res) = res else {
+            return dav1d_parse_sequence_header_error(res.map(|_| ()), c, &mut buf);
+        };
```

Comment:
```suggestion
        let res = match res {
            Ok(res) => res,
            Err(res) => return dav1d_parse_sequence_header_error(Err(res), c, &mut buf),
        }
```

Is this clearer?

Reply:
yes, this is much better, I didn't like that I had to call res.map(...) to convert between the error types. Thanks!